common/source/docs/common-blheli32-passthru.rst

```diff
@@ -1,62 +1,158 @@
+
+.. image:: ../../../images/blheli-version-check.png
+    :target: ../_images/blheli-version-check.png
+    :width: 450px
 
-Many brushless BLHeli ESCs offer telemetry reporting of important ESC data, such as RPM, Temperature, Current, etc. for use in OSDs or for controlling ArduPilot functions like variable center frequency noise filters. This can be via a serial connection to one of the autopilot's UART RX inputs, whose ``SERIALx_PROTOCOL`` has been set to "16" (ESC Telemetry), or directly over its control connection if it has Bi-Directional DShot capability (only available on certain BLHeli32 capable ESCs at present).
+Set the following parameters to enable BLHeli32 and BLHeli_S bi-directional Dshot:
 
-.. image:: ../../../images/dshot-telemwire.png
-    :target: https://shop.holybro.com/holybro-tekko32-esc35a_p1074.html
+- Set :ref:`SERVO_BLH_AUTO <SERVO_BLH_AUTO>` to 1 to automatically enable pass-through on all outputs configured as motors (e.g. :ref:`SERVOx_FUNCTION <SERVO9_FUNCTION>` = "Motor1", "Motor2", etc) for multicopters and quadplanes or throttle (e.g. those with :ref:`SERVOx_FUNCTION <SERVO9_FUNCTION>` set to 70 ("throttle"), 73 ("throttle left") or 74 ("throttle right")) on rovers.  For most multicopters, quadplanes and rovers this will do the right thing but for planes, you will need to further specify the used outputs as follows:
 
-*image courtesy of holybro.com*
+  - :ref:`SERVO_BLH_MASK <SERVO_BLH_MASK>` : a bitmap used to enable BLHeli32 pass-through and telemetry support on non-multirotor motors and / or exactly specify which servo outputs you want to enable pass-through and telemetry on (if available in ESC).
 
-For BLHeli32 ESC telemetery, see:
+  - :ref:`SERVO_BLH_OTYPE<SERVO_BLH_OTYPE>` : This needs to be set to the protocol being used for the DShot protocol being used on those additional outputs if not the same as the normal copter style motor outputs.
 
-.. toctree::
-    :maxdepth: 1
+- :ref:`SERVO_BLH_BDMASK <SERVO_BLH_BDMASK>` : a bitmap used to enable BLHeli32 or BLHeli_S bi-directional Dshot support. On flight controllers without IOMCU this would normally be set to 15 to indicate four active channels. On flight controllers with an IOMCU this can be set to 3840 to indicate four active AUX channels (bi-directional Dshot will only work on the AUX outputs). The BeastH7 only supports channels 1 and 4 for bi-directional dshot (mask set to 9).
 
-    BLHeli32 ESC Telemetry <common-dshot-blheli32-telemetry>
+- :ref:`SERVO_BLH_OTYPE<SERVO_BLH_OTYPE>` : This needs to be set to the protocol being used for the DShot protocol being used on those additional outputs if not the same as the normal copter style motor outputs.
```

Comment:
Only if MOT_PWM_TYPE is not set

Reply:
Txs.  I've removed the SERVO_BLH_OTYPE explanation because I think we only need to specify the setup required that is specific for the bi-directional dshot.  This parameter should have already been set as part of the regular DShot setup.